eth/api.go

```diff
@@ -1376,6 +1376,140 @@ func (api *PublicDebugAPI) SeedHash(number uint64) (string, error) {
 	return fmt.Sprintf("0x%x", hash), nil
 }
 
+
+type StructLogRes struct {
+	Pc      uint64            `json:"Pc"`
+	Op      string            `json:"Op"`
+	Gas     *big.Int          `json:Gas`
+	GasCost *big.Int          `json:GasCost`
+	Error   error             `json:Error`
+	Stack   []string          `json:"Stack"`
+	Memory  map[string]string `json:Memory`
+	Storage map[string]string `json:Storage`
+}
+
+type TransactionExecutionRes struct {
+	Gas         *big.Int       `json:Gas`
+	ReturnValue string         `json:ReturnValue`
+	StructLogs  []StructLogRes `json:StructLogs`
+}
+
+func (s *PublicDebugAPI) doReplayTransaction(txHash common.Hash) ([]vm.StructLog, []byte, *big.Int, error) {
```

Comment:
We're moving towards a direction where there will be publicly available parts of namespaces and privately available parts too. Although the design is still under debate, the idea is that anything that's publicly queryable should be in a public namespace and private or more sensitive things in the private namespace. Given that transaction execution could be potentially expensive computation wise, I think the private debug endpoint would be a better choice for it. Please move it into `PrivateDebugAPI`.